lib/experimental/full-page-client-side-navigation.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Registers full page client-side navigation option using the Interactivity API and adds the necessary directives.
+ */
+
+/**
+ * Enqueue the interactivity router script.
+ */
+function _gutenberg_enqueue_interactivity_router() {
+	// Set the navigation mode to full page client-side navigation.
+	wp_interactivity_config( 'core/router', array( 'navigationMode' => 'fullPage' ) );
+	wp_enqueue_script_module( '@wordpress/interactivity-router' );
+}
+
+add_action( 'wp_enqueue_scripts', '_gutenberg_enqueue_interactivity_router' );
+
+/**
+ * Set enhancedPagination attribute for query loop when the experiment is enabled.
+ *
+ * @param array $parsed_block The parsed block.
+ *
+ * @return array The same parsed block with the modified attribute.
+ */
+function _gutenberg_add_enhanced_pagination_to_query_block( $parsed_block ) {
+	if ( 'core/query' !== $parsed_block['blockName'] ) {
+		return $parsed_block;
+	}
+
+	$parsed_block['attrs']['enhancedPagination'] = true;
+	return $parsed_block;
+}
+
+add_filter( 'render_block_data', '_gutenberg_add_enhanced_pagination_to_query_block' );
+
+/**
+ * Add directives to all links.
+ *
+ * Note: This should probably be done per site, not by default when this option is enabled.
+ *
+ * @param array $content The block content.
+ *
+ * @return array The same block content with the directives needed.
+ */
+function _gutenberg_add_client_side_navigation_directives( $content ) {
+	$p = new WP_HTML_Tag_Processor( $content );
+	while ( $p->next_tag( array( 'tag_name' => 'a' ) ) ) {
+		if ( empty( $p->get_attribute( 'data-wp-on--click' ) ) ) {
+			$p->set_attribute( 'data-wp-on--click', 'core/router::actions.navigate' );
+		}
+		if ( empty( $p->get_attribute( 'data-wp-on--mouseenter' ) ) ) {
```

Comment:
What about mobile devices? Does `mouseenter` still work there somehow? Based on the event name I would think we need to also support a mobile friendly event.

Reply:
AFAIK no, `mouseenter` does not work on mobile. Closest thing would be `touchstart`.

Reply:
This is also a good example for where multiple actions should be able to be specified for a given event, as we've been discussing in https://github.com/WordPress/gutenberg/discussions/60574. There should not be a requirement that the `data-wp-on--mouseenter` attribute be empty. It should rather do something like:

```php
$p->set_attribute( 'data-wp-on--mouseenter--core-router', 'core/router::actions.prefetch' );
```

Reply:
Maybe let's wait until #60661 has landed, then we can fix these attributes to be unconditionally added, using a suffix for uniqueness.

Reply:
I added a follow-up task to [this list](https://github.com/WordPress/gutenberg/pull/59707#issuecomment-2058832203) to explore and decide how to interact with mobile events.